cli/state.rs

```diff
@@ -195,158 +163,74 @@ impl Loader for ThreadSafeState {
   ) -> Box<deno::SourceCodeInfoFuture> {
     self.metrics.resolve_count.fetch_add(1, Ordering::SeqCst);
     let module_url_specified = module_specifier.to_string();
-    Box::new(self.fetch_compiled_module(module_specifier).map(
-      |compiled_module| deno::SourceCodeInfo {
+    let fut = self
+      .global_state
+      .fetch_compiled_module(module_specifier)
+      .map(|compiled_module| deno::SourceCodeInfo {
         // Real module name, might be different from initial specifier
         // due to redirections.
         code: compiled_module.code,
         module_url_specified,
         module_url_found: compiled_module.name,
-      },
-    ))
+      });
+
+    Box::new(fut)
   }
 }
 
 impl ThreadSafeState {
   pub fn new(
-    flags: flags::DenoFlags,
-    argv_rest: Vec<String>,
-    progress: Progress,
+    global_state: ThreadSafeGlobalState,
+    main_module: Option<ModuleSpecifier>,
     include_deno_namespace: bool,
   ) -> Result<Self, ErrBox> {
-    let custom_root = env::var("DENO_DIR").map(String::into).ok();
-
     let (worker_in_tx, worker_in_rx) = async_mpsc::channel::<Buf>(1);
     let (worker_out_tx, worker_out_rx) = async_mpsc::channel::<Buf>(1);
-    let internal_channels = (worker_out_tx, worker_in_rx);
-    let external_channels = (worker_in_tx, worker_out_rx);
-    let resource = resources::add_worker(external_channels);
-
-    let dir = deno_dir::DenoDir::new(custom_root)?;
-
-    let file_fetcher = SourceFileFetcher::new(
-      dir.deps_cache.clone(),
-      progress.clone(),
-      !flags.reload,
-      flags.cache_blacklist.clone(),
-      flags.no_fetch,
-    )?;
-
-    let ts_compiler = TsCompiler::new(
-      file_fetcher.clone(),
-      dir.gen_cache.clone(),
-      !flags.reload,
-      flags.config_path.clone(),
-    )?;
-
-    let main_module: Option<ModuleSpecifier> = if argv_rest.len() <= 1 {
-      None
-    } else {
-      let root_specifier = argv_rest[1].clone();
-      Some(ModuleSpecifier::resolve_url_or_path(&root_specifier)?)
+    let internal_channels = WorkerChannels {
+      sender: worker_out_tx,
+      receiver: worker_in_rx,
     };
-
-    let import_map: Option<ImportMap> = match &flags.import_map_path {
-      None => None,
-      Some(file_path) => Some(ImportMap::load(file_path)?),
+    let external_channels = WorkerChannels {
+      sender: worker_in_tx,
+      receiver: worker_out_rx,
     };
 
-    let mut seeded_rng = None;
-    if let Some(seed) = flags.seed {
-      seeded_rng = Some(Mutex::new(StdRng::seed_from_u64(seed)));
-    };
+    let mut table = resources::lock_resource_table();
+    let rid = table.add(Box::new(external_channels));
 
-    let modules = Arc::new(Mutex::new(deno::Modules::new()));
+    let import_map: Option<ImportMap> =
+      match global_state.flags.import_map_path.as_ref() {
+        None => None,
+        Some(file_path) => Some(ImportMap::load(file_path)?),
```

Comment:
Does this mean the import map is loaded every time a worker is created? (I note this is the behavior beforehand too - just curious)

Reply:
Yes, this bit is to be removed for "web workers". Only main worker should have import map applied. Will open an issue for that